src/Ocean/OceanProblems/initial_value_problem.jl

```diff
@@ -0,0 +1,99 @@
+#####
+##### Initial value problem
+#####
+
+struct InitialValueProblem{FT, IC, BC} <: AbstractSimpleBoxProblem
+    Lˣ::FT
+    Lʸ::FT
+    H::FT
+    initial_conditions::IC
+    boundary_conditions::BC
+end
+
+"""
+    InitialValueProblem(FT=Float64; dimensions, initial_conditions=InitialConditions(),
+                        boundary_conditions = (OceanBC(Impenetrable(FreeSlip()), Insulating()),
+                                               OceanBC(Penetrable(FreeSlip()), Insulating())))
+
+Returns an `InitialValueProblem` with `dimensions = (Lˣ, Lʸ, H)`, `initial_conditions`,
+and `boundary_conditions`.
+
+The default `initial_conditions` are resting with no temperature perturbation;
+the default `boundary_conditions` are horizontally-periodic with `Insulating()`
+and `FreeSlip()` conditions at the top and bottom.
+"""
+function InitialValueProblem(
+    FT = Float64;
```

Comment:
Can we make `FT` non-optional. This could hide type instabilities.

Reply:
I can do that here. I am submitting a PR soon that may relieve the user from concerns like these by introducing a single location for specifying the type of participants in the Ocean's `BalanceLaw`, however. If the design proposed by that PR is accepted, `InitialValueProblem` may eventually become obsolete.

Reply:
@charleskawczynski the type here is probably unnecessary because they only specify the type of the domain limits. The domain limits are used here:

https://github.com/CliMA/ClimateMachine.jl/blob/4a3f08659b23e5be5d0a83ce72ad8c33df824ea2/src/Driver/driver_configs.jl#L309-L327

and it looks like a variable called `FloatType` is passed to `DiscontinuousSpectralElementGrid`. Is `FloatType` respected regardless of the eltype of `brickrange`?

If it doesn't affect type instability, is it okay to relieve the user of the need to explicitly type this object?

Reply:
I made the changes anyways.